controller/src/main/java/io/pravega/controller/store/PravegaTablesScope.java

```diff
@@ -219,10 +219,12 @@ public void refresh() {
     public CompletableFuture<Void> addReaderGroupToScope(String readerGroupName, UUID readerGroupId) {
         return getReaderGroupsInScopeTableName()
                 .thenCompose(tableName -> Futures.toVoid(Futures.exceptionallyComposeExpecting(
-                        storeHelper.addNewEntry(tableName, readerGroupName, getIdInBytes(readerGroupId)),
+                        storeHelper.addNewEntryIfAbsent(tableName, readerGroupName, getIdInBytes(readerGroupId)),
                         e -> Exceptions.unwrap(e) instanceof StoreException.DataNotFoundException,
                         () -> storeHelper.createTable(tableName)
-                                .thenCompose(v -> storeHelper.addNewEntry(tableName, readerGroupName, getIdInBytes(readerGroupId))))));
+                                .thenCompose(v -> storeHelper.addNewEntry(tableName, readerGroupName,
```

Comment:
shouldnt this also be addNewEntryIfAbsent?

Reply:
Good catch. Yes, will change.

Reply:
fixed